core/src/serve_repair.rs

```diff
@@ -196,13 +198,39 @@ impl ServeRepair {
         requests_receiver: &PacketReceiver,
         response_sender: &PacketSender,
         stats: &mut ServeRepairStats,
+        max_packets: &mut usize,
     ) -> Result<()> {
         //TODO cache connections
         let timeout = Duration::new(1, 0);
-        let reqs = requests_receiver.recv_timeout(timeout)?;
-        stats.total_packets += reqs.packets.len();
+        let mut reqs_v = vec![requests_receiver.recv_timeout(timeout)?];
+        let mut total_packets = reqs_v[0].packets.len();
+
+        let mut dropped_packets = 0;
+        while let Ok(more) = requests_receiver.try_recv() {
+            total_packets += more.packets.len();
```

Comment:
Just double checking, but each of these packets is at most 256 in length right?

Because we increment first here before we drop, worst case scenario we have 769 packets, and then get a batch of 256 which get dropped, even though ideally we could have kept 255 more of them and stayed below the limit.